src/gas-price/gas-price.ts

```diff
@@ -0,0 +1,236 @@
+import { go } from '@api3/promise-utils';
+import { ethers } from 'ethers';
+
+import { loadConfig } from '../config';
+import type { GasSettings } from '../config/schema';
+
+interface DataFeedValue {
+  value: ethers.BigNumber;
+  timestamp: number;
+}
+interface GasState {
+  gasPrices: { price: ethers.BigNumber; timestamp: number }[];
+  lastOnChainDataFeedValues: Record<string, DataFeedValue>;
+}
+
+export const gasPriceStore: Record<string, Record<string, GasState>> = {};
+
+export const initializeGasStore = (chainId: string, providerName: string) => {
+  if (!gasPriceStore[chainId]) {
+    gasPriceStore[chainId] = {};
+  }
+
+  if (!gasPriceStore[chainId]![providerName]) {
+    gasPriceStore[chainId]![providerName] = { gasPrices: [], lastOnChainDataFeedValues: {} };
+  }
+};
+
+/**
+ * Saves a gas price into the store.
+ * @param chainId
+ * @param providerName
+ * @param gasPrice
+ */
+export const setStoreGasPrices = (chainId: string, providerName: string, gasPrice: ethers.BigNumber) => {
+  gasPriceStore[chainId]![providerName]!.gasPrices = [
+    { price: gasPrice, timestamp: Date.now() },
```

Comment:
When we say `timestamp` we usually refer to UNIX timestamp (in seconds). I'd rename this one to `timestampMs` just to be explicit or convert to seconds.